### PR TITLE
Wrangling with Iced

### PR DESCRIPTION
**Pull request by geralddejong**
Layout etc

### CODE REVIEW

File: src/gui.rs [hunk 1]
```diff
@@ -36,7 +35,7 @@ pub struct GUI {
 impl GUI {
     pub fn new(graphics: &GraphicsWindow, window: &Window) -> Self {
         let viewport = Viewport::with_physical_size(
-            Size::new(1600, 1200),
+            Size::new(graphics.size.width, graphics.size.height),
```
Comment (mitchelljustin) [inline src/gui.rs:38]: Ooh nice

File: src/gui.rs [hunk 2]
```diff
@@ -93,7 +92,11 @@ impl GUI {
     pub fn post_render(&mut self) {
         self.staging_belt.recall();
     }
-
+    
+    pub fn change_state(&mut self, message: Message) {
```
Comment (mitchelljustin) [inline src/gui.rs:96]: Why not just name this `queue_message` also?

Comment (geralddejong) [reply src/gui.rs:96]: That wouldn't indicate what the message is for.

File: src/gui.rs [hunk 3]
```diff
@@ -193,7 +206,9 @@ pub enum Message {
 impl Default for Controls {
     fn default() -> Self {
         Self {
+            showing: Showing::Nothing,
             measure_nuance: 0.0,
+            lower_limit: f32::MIN,
```
Comment (mitchelljustin) [inline src/gui.rs:211]: I prefer more explicit: `strain_lower_limit`

File: src/gui.rs [hunk 4]
```diff
@@ -232,31 +257,52 @@ impl Program for Controls {
...
+                .align_items(Alignment::End)
+                .push(
+                    Row::new()
+                        .height(Length::Fill)
+                        .width(Length::Fill)
+                        .push(right_column)
+                )
+                .push(
+                    match self.showing {
+                        Showing::Nothing => Row::new(),
```
Comment (mitchelljustin) [inline src/gui.rs:280]: No button to set StrainThreshold panel to shown?

File: src/gui.rs [hunk 5]
```diff
@@ -232,31 +257,52 @@ impl Program for Controls {
...
+                        Showing::StrainThreshold => {
+                            Row::new()
+                                .padding(20)
+                                .spacing(20)
+                                .push(
+                                    Text::new("Strain threshold")
```
Comment (mitchelljustin) [inline src/gui.rs:286]:
```suggestion
                                    Text::new("Strain Threshold")
```

File: src/gui.rs [hunk 6]
```diff
@@ -232,31 +257,52 @@ impl Program for Controls {
...
+                                )
+                                .push(
+                                    Button::new(Text::new("Add Pulls"))
+                                        .on_press(Message::AddPulls)
+                                )
+                        }
+                    }
+                )
+                .into();
+        // element.explain(Color::WHITE)
```
Comment (mitchelljustin) [inline src/gui.rs:305]: ?

Comment (mitchelljustin) [reply src/gui.rs:305]: Oh that's for the debug outline

File: src/gui.rs (hunk 3 above)
Comment (geralddejong) [reply src/gui.rs:211]: proly right. on it.

File: src/application.rs [hunk 7]
```diff
@@ -143,33 +144,36 @@ pub fn run() {
...
+                window.set_cursor_icon(app.gui.cursor_icon());
             }
             Event::MainEventsCleared => {
-                application.gui.update();
-                for action in application.gui.controls().take_actions() {
+                app.gui.update();
+                for action in app.gui.controls().take_actions() {
                     match action {
-                        Action::AddPulls { measure_nuance } => {
+                        Action::AddPulls { strain_nuance: measure_nuance } => {
```
Comment (mitchelljustin) [inline src/application.rs:176]: Just rename the local variable to `strain_nuance` as well?

**Pull request merged**